release.rb

```diff
@@ -0,0 +1,215 @@
+class ReleaseScript
+
+  def self.run(version, release_notes_file_path)
+    run_internal(version, release_notes_file_path, false)
+  end
+
+  def self.run_force(version, release_notes_file_path)
+    run_internal(version, release_notes_file_path, true)
+  end
+
+  private
+
+  def self.run_internal(version, release_notes_file_path, force)
+    remote_branch = 'convert_release_script_to_rake_task'
+    pod_name = 'Quick'
+    pod_spec = 'Quick.podspec'
+    github_tags_url = 'https://github.com/jwfriese/Quick/tags'
+    carthage_framework_name = 'Quick'
+    carthage = ENV["CARTHAGE"] || 'carthage'
+    pod = ENV["COCOAPODS"] || 'pod'
+    version_tag = "v#{version}"
+
+    puts 'Verifying Local Directory for Release'
+    carthage_available = verify_availability(carthage) 
+    die("Carthage is required to produce a release. Aborting.") unless carthage_available
+    puts ' > Carthage is installed'
+
+    pod_available = verify_availability(pod)
+    die(" Cocoapods is required to produce a release. Aborting.") unless pod_available
+    puts ' > Cocoapods is installed'
+
+    puts ' > Is this a reasonable tag?'
+
+    verify_version(version_tag, force)
+
+    release_notes = gather_release_notes(release_notes_file_path, version, pod_name)
+    puts " > Release notes: #{release_notes}"
+
+    die("Cannot find podspec: #{pod_spec}. Aborting.") unless File.file?("#{pod_spec}")
+
+    puts ' > Podspec exists'
+    
+    pgp_found = system("git config --get user.signingkey > /dev/null")
+    unless pgp_found
+      puts '[ERROR] No PGP found to sign tag. Aborting.'
+      puts
+      puts '  Creating a release requires signing the tag for security purposes. This allows users to verify the git cloned tree is from a trusted source.'
+      puts '  From a security perspective, it is not considered safe to trust the commits (including Author & Signed-off fields). It is easy for any'
+      puts '  intermediate between you and the end-users to modify the git repository.'
+      puts
+      puts "  While not all users may choose to verify the PGP key for tagged releases. It is a good measure to ensure 'this is an official release'"
+      puts '  from the official maintainers.'
+      puts
+      puts "  If you're creating your PGP key for the first time, use RSA with at least 4096 bits."
+      puts
+      puts 'Related resources:'
+      puts ' - Configuring your system for PGP: https://git-scm.com/book/tr/v2/Git-Tools-Signing-Your-Work'
+      puts ' - Why: http://programmers.stackexchange.com/questions/212192/what-are-the-advantages-and-disadvantages-of-cryptographically-signing-commits-a'
+      puts
+      return
+    end
+
+    puts " > Found PGP key for git"
+
+    #verify_cocoapods_trunk_ownership(pod_name)
+    #puts " > Verified ownership to #{pod_name} pod"
```

Comment:
As noted in the leading comment, these validation steps were always failing for me because the ```pod trunk me``` command did not pass Quick back as one of my pods.